### ISSUE TITLE
Absolute base url calculated relative to running code, not tsconfig path

### ISSUE THREAD

**Issue by artaommahe** [jonaskello/tsconfig-paths#7]
I have this app structure
```
tsconfig.json
shared/
services/
  service1/
    index.js
```
with
```
const tsConfig = require('../../tsconfig.json');
const tsConfigPaths = require('tsconfig-paths');

tsConfigPaths.register({
  baseUrl: tsConfig.compilerOptions.baseUrl,
  paths: tsConfig.compilerOptions.paths
});
```
inside index.js
tsconfig basUrl configured as
```
    "baseUrl": "./services/",
    "shared/*": [ "../shared/*" ]
```
to allow imports like `import {} from 'service1/smth'`

but `tsconfig-paths` makes wrong absolute path for this baseUrl
```
configLoaderResult { resultType: 'success',
  absoluteBaseUrl: '.../services/service1/services/',
```
instead of just `.../services/`

https://github.com/jonaskello/tsconfig-paths/blob/master/src/config-loader.ts#L35

**Comment by artaommahe** [comment]
Also looks like there is an issue with matching paths related to baseUrl.
I passed absolute url as baseUrl to `tsConfigPaths.register()` (to bypass first issue) and still have an error
```
import {  } from 'service1/client';
...
Error: Cannot find module 'service1/client'
```
(client placed right in `service1` folder)
debugging led me to `console.log(absolutePaths);` here https://github.com/jonaskello/tsconfig-paths/blob/master/src/match-path.ts#L23 that prints only this
```
{ 'shared/*': [ '.../shared/*' ] }
```
So no one path related to baseUrl will be handled except explicitly added `shared`.

IDE (VSCode) works well with config above

**Comment by Jontem** [comment]
Could you create a repro that shows your problem? It's easier to debug something that is runnable.

**Comment by artaommahe** [comment]
@Jontem i tried to reproduce same folders/package.json structure here
https://github.com/artaommahe/tsconfig-paths-7
clone, run
```
yarn
cd services/smth
yarn run start
```

see an error
```
Error: Cannot find module 'user/user'
```

then
```
// add to tsconfig.json
    "paths": {
      "*": [ "./*" ],

// uncomment in services/smth/index.js
  baseUrl: path.join(__dirname, '../../', tsConfig.compilerOptions.baseUrl),
  /*baseUrl: tsConfig.compilerOptions.baseUrl,*/
```
and this will work
Same error if you comment out `import { user } from 'user/user';`
```
Error: Cannot find module 'shared/shared'
```
fixes with passing absolute `baseUrl`

**Comment by Jontem** [comment]
First if you specify a relative baseUrl it will be resolved relative to where tsconfig.json is. As stated in README.

Second your path mappings will work if looking like this

```ts
  paths: {
    "*": ["./*"],
    "shared/*": ["../shared/*"]
  }
```

**Comment by artaommahe** [comment]
> First if you specify a relative baseUrl it will be resolved relative to where tsconfig.json is. As stated in README.

yep, and it's correclty mapped to `./services/` but `import ... from 'user/*'` does not work.

> Second your path mappings will work if looking like this

yep, i wrote this at second post, but it looks ugly - typescript correctly understand paths related to `baseUrl` without this `paths` hack but `tsconfig-paths` not.

**Comment by Jontem** [comment]
So what you mean is that you would like tsconfig-paths to resolve modules relative to baseUrl without having to specify a path mapping explicitly?

**Comment by artaommahe** [comment]
@Jontem yep, imho `baseUrl` should be added as first path for resolving modules imports

**Comment by Jontem** [comment]
I've made an implementation of this in the commit https://github.com/jonaskello/tsconfig-paths/commit/cb18815b80374a6c071061bb8d92fc52b35c69e0

I have just quickly tested it and it seems to work. I'll release it when i have more time to verify. But please try it out.

**Comment by artaommahe** [comment]
@Jontem it works, thx

**Comment by Jontem** [comment]
Fixed in 2.2.0
